src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -360,6 +360,18 @@ internal async Task<IActorReminder> RegisterReminderAsync(ActorReminderOptions o
             return reminder;
         }
 
+        /// <summary>
+        /// Gets a reminder previously registered using <see cref="Dapr.Actors.Runtime.Actor.RegisterReminderAsync(ActorReminderOptions)" />.
+        /// </summary>
+        /// <param name="reminderName">The name of the reminder to get.</param>
+        /// <returns>
+        /// Returns a task that represents the asynchronous get operation. The result of the task contains the reminder if it exists, otherwise null.
+        /// </returns>
+        protected async Task<IActorReminder> GetReminderAsync(string reminderName)
```

Comment:
This method states it returns `null` if no timer exists, but the underlying call to the `DaprHttpInteractor` appears to throw if the HTTP request does not return a successful response (and the docs say the Actor Get Reminder API returns only `200` and `500` responses.

Reply:
In case the reminder doesn't exist, the HTTP call TO Dapr returns a successful status code with a response {"null"}. This is being handled in the `DeserializeReminderAsync` function in ActorTimerManager(https://github.com/dapr/dotnet-sdk/pull/1103/files#diff-89d1bcb06652638fdfeaa1b8146ada6ea2d8fe4ff18216de0928734712473cf4R103), in which we actually return a null object. Also the same scenario is added in E2E test as well, testing both cases when the reminder doesn't exist and when reminder exists.
The E2E test case for no reminder can be found here: https://github.com/dapr/dotnet-sdk/pull/1103/files#diff-6c279ec059b177b7a7e54e34eda712f299033fb2321fac66836f93a78c76b5d5R62

Reply:
If the reminder doesn't exist, Dapr returns a suceessful status code with {null} response. That scenario is already handled in Json Serializer and so in case a reminder doesn't exist, this function returns a null value only.

Alos, the same has been tested in added E2E test as well and it is passing as well: https://github.com/dapr/dotnet-sdk/pull/1103/files#diff-6c279ec059b177b7a7e54e34eda712f299033fb2321fac66836f93a78c76b5d5R61